### DISCUSSION TITLE
Support to Pytorch 1.9

### DISCUSSION THREAD

**Discussion by Jingnan-Jia** [Project-MONAI/MONAI#2471] [General]
Just a reminder:

Some APIs need to be updated to support Pytorch 1.9. For instance: [torch.testing.assert_allclose](https://github.com/Project-MONAI/MONAI/blob/91ea1b676a8fda066fd07ba6b55e87151da16dde/tests/test_activations.py#L73) should be replaced by [torch.testing.assert_close](https://pytorch.org/docs/stable/testing.html).

And pytorch 1.9 has already [Added Numpy seeding to worker of DataLoader](https://www.apowersoft.cn/music-free-download-webs.html#:~:text=%E8%99%BE%E7%B1%B3%E3%80%81%E9%85%B7%E7%8B%97%E3%80%81%E9%85%B7%E6%88%91%E3%80%81%E5%92%AA%E5%92%95%E7%AD%89%E5%87%A0%E5%8D%81%E4%B8%AA%E9%9F%B3%E4%B9%90%E8%BD%AF%E4%BB%B6%E7%9A%84%E6%90%9C%E7%B4%A2%E7%B1%BB%E7%BD%91%E7%AB%99%2C%E5%8F%AA%E8%A6%81%E5%9C%A8%E6%90%9C%E7%B4%A2%E6%A0%8F%E4%B8%8B%E8%BE%93%E5%85%A5%E6%AD%8C%E6%9B%B2%E5%90%8D%2C%E4%BC%9A%E7%9B%B4%E6%8E%A5%E5%B0%86%E6%90%9C%E7%B4%A2%E5%88%B0%E7%9A%84%E7%9B%B8%E5%85%B3%E6%AD%8C%E6%9B%B2%E5%85%A8%E9%83%A8%E5%B1%95%E7%A4%BA) ...... **users now won’t be required to set random seed for NumPy using worker_init_fn to force NumPy random operations deterministic and independent across DataLoader workers.**. So the `monai.data.dataloader` seems not necessary anymore?

**Comment by Nic-Ma** [comment]
Hi @Jingnan-Jia ,

Thanks for your interest here.
1. Currently, we are trying to support PyTorch 1.5 - 1.9, so we should make MONAI APIs compatible with these versions.
2. `monai.data.DataLoader` also uses our enhanced `collate_fn`.

Thanks.
